timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -324,18 +367,42 @@ def predict(
         """
         if self.target_scaler is not None:
             data = self.target_scaler.fit_transform(data)
+        if self.covariates_regressor is not None:
+            if self.covariates_regressor.refit_during_predict:
+                data = self.covariates_regressor.fit_transform(data)
+            else:
+                data = self.covariates_regressor.transform(data)
 
         data = self.preprocess(data, is_train=False)
         known_covariates = self.preprocess_known_covariates(known_covariates)
+
+        # FIXME: Avoid copying covariates_regressor across processes if self._predict uses joblib.
+        # FIXME: The clean solution is to convert all methods executed in parallel to @classmethod
+        covariates_regressor = self.covariates_regressor
+        self.covariates_regressor = None
         predictions = self._predict(data=data, known_covariates=known_covariates, **kwargs)
```

Comment:
Yes, the `_predict` method will expect the covariates to be present in accordance with the `CovariateMetadata` (and the `_predict` method ideally shouldn't be aware that a "regressor" object even exists). It will mostly be user's and our responsibility not to do silly things such as using both the "regressor" in combination with a model that natively handles the covariates.